File: BHAVRP_Python/cujae/inf/citi/om/factory/interfaces/assignment_type.py
from enum import Enum

# Enumerado que indica los tipos de métodos de asignación.
class AssignmentType(Enum):
    BestCyclicAssignment = "cujae.inf.citi.om.assignment.others.basedcyclic.best_cyclic_assignment"
    
    BestNearest = "cujae.inf.citi.om.assignment.others.distance.best_nearest"
    
    Clara = "cujae.inf.citi.om.assignment.clustering.partitional.clara"
    
    CoefficientPropagation = "cujae.inf.citi.om.assignment.classical.cluster.coefficient_propagation"
    
    CyclicAssignment = "cujae.inf.citi.om.assignment.classical.cyclic.cyclic_assignment"
    
    FarthestFirst = "cujae.inf.citi.om.assignment.clustering.partitional.farthest_first"
    
    Kmeans = "cujae.inf.citi.om.assignment.clustering.partitional.k_means"
    
    NearestByCustomer = "cujae.inf.citi.om.assignment.others.distance.nearest_by_customer"
    
    NearestByDepot = "cujae.inf.citi.om.assignment.others.distance.nearest_by_depot"
    
    PAM = "cujae.inf.citi.om.assignment.clustering.partitional.pam"
    
    Parallel = "cujae.inf.citi.om.assignment.classical.urgency.parallel"
    
    RandomByElement = "cujae.inf.citi.om.assignment.others.distance.random_by_element"
    
    RandomSequentialCyclic = "cujae.inf.citi.om.assignment.others.basedcyclic.random_sequential_cyclic"
    
    SequentialCyclic = "cujae.inf.citi.om.assignment.others.basedcyclic.sequential_cyclic"
    
    Simplified = "cujae.inf.citi.om.assignment.classical.urgency.simplified"
    
    Sweep = "cujae.inf.citi.om.assignment.classical.urgency.sweep"
    
    ThreeCriteriaClustering = "cujae.inf.citi.om.assignment.classical.cluster.three_criteria_clustering"
    
    UPGMC = "cujae.inf.citi.om.assignment.clustering.hierarchical.upgmc"
    
    def __str__(self):
        return self.value

File: BHAVRP_Java/BHAVRPv2.0/BHAVRP2.0/src/cujae/inf/ic/om/factory/interfaces/AssignmentType.java
package cujae.inf.ic.om.factory.interfaces;

/* Enumerado que indica los tipos de métodos de asignación*/
public enum AssignmentType {

	BestCyclicAssignment	
	{
		@Override
		public String toString()
		{
			return cujae.inf.ic.om.assignment.others.basedcyclic.BestCyclicAssignment.class.getName(); 
		}
	}, 

	BestNearest
	{
		@Override
		public String toString()
		{
			return cujae.inf.ic.om.assignment.others.distance.BestNearest.class.getName(); 
		}
	}, 

	CLARA
	{
		@Override
		public String toString()
		{
			return cujae.inf.ic.om.assignment.clustering.partitional.Clara.class.getName(); 
		}
	}, 
	
	CoefficientPropagation
	{
		@Override
		public String toString()
		{
			return cujae.inf.ic.om.assignment.classical.cluster.CoefficientPropagation.class.getName(); 
		}
	}, 

	CyclicAssignment
	{
		@Override
		public String toString()
		{
			return cujae.inf.ic.om.assignment.classical.cyclic.CyclicAssignment.class.getName(); 
		}
	}, 

	Farthest_First
	{
		@Override
		public String toString()
		{
			return cujae.inf.ic.om.assignment.clustering.partitional.Farthest_First.class.getName(); 
		}
	},
	
	KMEANS
	{
		@Override
		public String toString()
		{
			return cujae.inf.ic.om.assignment.clustering.partitional.Kmeans.class.getName(); 
		}
	}, 
	
	NearestByCustomer
	{
		@Override
		public String toString()
		{
			return cujae.inf.ic.om.assignment.others.distance.NearestByCustomer.class.getName(); 
		}
	}, 
	
	NearestByDepot
	{
		@Override
		public String toString()
		{
			return cujae.inf.ic.om.assignment.others.distance.NearestByDepot.class.getName(); 
		}
	}, 
	
	PAM
	{
		@Override
		public String toString()
		{
			return cujae.inf.ic.om.assignment.clustering.partitional.PAM.class.getName(); 
		}
	}, 
	
	Parallel
	{
		@Override
		public String toString()
		{
			return cujae.inf.ic.om.assignment.classical.urgency.Parallel.class.getName(); 
		}
	}, 
	
	RandomByElement
	{
		@Override
		public String toString()
		{
			return cujae.inf.ic.om.assignment.others.distance.RandomByElement.class.getName(); 
		}
	}, 

	RandomSequentialCyclic
	{
		@Override
		public String toString()
		{
			return cujae.inf.ic.om.assignment.others.basedcyclic.RandomSequentialCyclic.class.getName(); 
		}
	}, 
	
	SequentialCyclic
	{
		@Override
		public String toString()
		{
			return cujae.inf.ic.om.assignment.others.basedcyclic.SequentialCyclic.class.getName(); 
		}
	}, 

	Simplified
	{
		@Override
		public String toString()
		{
			return cujae.inf.ic.om.assignment.classical.urgency.Simplified.class.getName(); 
		}
	}, 
	
	Sweep
	{
		@Override
		public String toString()
		{
			return cujae.inf.ic.om.assignment.classical.urgency.Sweep.class.getName(); 
		}
	}, 
	
	ThreeCriteriaClustering
	{
		@Override
		public String toString()
		{
			return cujae.inf.ic.om.assignment.classical.cluster.ThreeCriteriaClustering.class.getName(); 
		}
	}, 

	UPGMC
	{
		@Override
		public String toString()
		{
			return cujae.inf.ic.om.assignment.clustering.hierarchical.UPGMC.class.getName(); 
		}
	};
}